ironfish/src/wallet/wallet.ts

```diff
@@ -680,6 +681,28 @@ export class Accounts {
     try {
       this.assertHasAccount(sender)
 
+      // check if the chain data is fully synced
+      let headHash = this.headHashes.get(sender.id)
+      headHash = await sender.getHeadHash()
+      if (!headHash) {
+        headHash = this.headHashes.get(sender.id)
+      }
+
+      if (!headHash) {
+        throw new ResponseError(`No headhash found for this account.`, ERROR_CODES.ERROR, 400)
+      }
+
+      const header = await this.chain.getHeader(headHash)
+      Assert.isNotNull(header, `Missing block header for hash '${headHash.toString('hex')}'`)
+
+      if (header.timestamp.valueOf() < Date.now() - MAX_SYNCED_AGE_MS) {
+        throw new ResponseError(
```

Comment:
nit - we use `ValidationError` elsewhere in this file already. It's mostly for convenience, but pre-fills the status and error code

Reply:
Do we want to consider this error as Validation error ?  Its not exact validation error since user request/input does not have issue.  but it's kind of fall into the category of blockchain not in valid state for creating transaction.